RockLib.Configuration.ObjectFactory/ConfigReloadingProxy.cs

```diff
@@ -9,198 +9,211 @@
+      {
+         if (typeof(TInterface) == typeof(IEnumerable))
+            throw new InvalidOperationException("The IEnumerable interface is not supported.");
+         if (typeof(IEnumerable).IsAssignableFrom(typeof(TInterface)))
+            throw new InvalidOperationException($"Interfaces that inherit from IEnumerable are not suported: '{typeof(TInterface).FullName}'");
+
+         _section = section ?? throw new ArgumentNullException(nameof(section));
+         _defaultTypes = defaultTypes ?? ConfigurationObjectFactory.EmptyDefaultTypes;
+         _valueConverters = valueConverters ?? ConfigurationObjectFactory.EmptyValueConverters;
+         _declaringType = declaringType; // Null is a valid value
+         _memberName = memberName; // Null is a valid value
+         _resolver = resolver ?? Resolver.Empty;
+         _hash = GetHash();
+         Object = CreateObject();
+         ChangeToken.OnChange(section.GetReloadToken, () => ReloadObject(false));
+      }
+
+      /// <summary>
+      /// Force the underlying object to reload from the current configuration.
+      /// </summary>
+      public void Reload() => ReloadObject(true);
+
+      /// <summary>
+      /// Gets the underlying object.
+      /// </summary>
+      [DebuggerBrowsable(DebuggerBrowsableState.RootHidden)]
+#pragma warning disable CA1720 // Identifier contains type name
+      public TInterface Object { get; private set; }
+#pragma warning restore CA1720 // Identifier contains type name
+
+      /// <summary>
+      /// Occurs immediately before the underlying object is reloaded.
+      /// </summary>
+      public event EventHandler? Reloading;
+
+      /// <summary>
+      /// Occurs immediately after the underlying object is reloaded.
+      /// </summary>
+      public event EventHandler? Reloaded;
+
+      /// <summary>
+      /// Dispose the underlying object if it implements <see cref="IDisposable"/>.
+      /// </summary>
+#pragma warning disable CA1063 // Implement IDisposable Correctly
+#pragma warning disable CA1816 // Dispose methods should call SuppressFinalize
+      public void Dispose() => (Object as IDisposable)?.Dispose();
+#pragma warning restore CA1816 // Dispose methods should call SuppressFinalize
+#pragma warning restore CA1063 // Implement IDisposable Correctly
+
+      private TInterface CreateObject()
+      {
+         // In order to create the object (and avoid infinite recursion), we need to figure out
+         // the concrete type to create and the config section that defines the value to create.
+         Type? concreteType;
+         IConfiguration valueSection;
+
+         // If _section contains a type-specified value, extract the type and use the value sub-section.
+         string typeValue = _section[ConfigurationObjectFactory.TypeKey];
+         if (typeValue != null)
```

Comment:
It looks like in some places == null or != null was updated to is null or is not null, but it looks like some might have been missed.

Reply:
We can use `is null`, but not `is not null`. That won't work for all target frameworks.k

FWIW I just did another search and I didn't find any `== null` in the code.